src/model/page-config-resolver-test.js

```diff
@@ -268,7 +272,337 @@ describes.realWin('PageConfigResolver', {}, env => {
+      addMicrodata(divElement);
+      readyState = 'complete';
+      return new PageConfigResolver(gd).resolveConfig().then(config => {
+        expect(config.isLocked()).to.be.false;
+        expect(config.getProductId()).to.equal('pub1:premium');
+      });
+    });
+
+    it('malformed microdata tree product info under section type', () => {
+      const divElement = createElement(doc, 'div');
+      divElement.innerHTML =
+        '<div itemscope itemtype="http://schema.org/NewsArticle"> \
+          <div itemscope itemtype="http://schema.org/Section"> \
+              <div itemprop="isPartOf" itemscope itemtype="http://schema.org/CreativeWork http://schema.org/Product"> \
+                <meta itemprop="name" content="New York Times"/> \
+                <meta itemprop="productID" content="pub1:premium"/> \
+              </div> \
+            </div> \
+            <div itemprop="hasPart" itemscope itemtype="http://schema.org/WebPageElement"> \
+              <meta itemprop="isAccessibleForFree" content="False"/> \
+              <meta itemprop="cssSelector" content=".paywalled-section"/> \
+            </div> \
+            <div itemprop="articleBody" class="paywalled-section"> \
+              Paid content possibly. \
+            </div> \
+          </div>';
+      addMicrodata(divElement);
+      readyState = 'complete';
+      const resolver = new PageConfigResolver(gd);
+      expect(resolver.check()).to.be.null;
+    });
+
+    it('multiple product info but one valid', () => {
+      const divElement = createElement(doc, 'div');
+      divElement.innerHTML =
+          '<div itemscope itemtype="http://schema.org/NewsArticle"> \
+          <div itemscope itemtype="http://schema.org/Section"> \
+              <div itemprop="isPartOf" itemscope itemtype="http://schema.org/CreativeWork http://schema.org/Product"> \
+                <meta itemprop="name" content="New York Times"/> \
+                <meta itemprop="productID" content="pub1:basic"/> \
+              </div> \
+            </div> \
+            <div itemprop="isPartOf" itemscope itemtype="http://schema.org/CreativeWork http://schema.org/Product"> \
+              <meta itemprop="name" content="New York Times"/> \
+              <meta itemprop="productID" content="pub1:premium"/> \
+            </div> \
+            <div itemprop="hasPart" itemscope itemtype="http://schema.org/WebPageElement"> \
+              <meta itemprop="isAccessibleForFree" content="False"/> \
+              <meta itemprop="cssSelector" content=".paywalled-section"/> \
+            </div> \
+            <div itemprop="articleBody" class="paywalled-section"> \
+              Paid content possibly. \
+            </div> \
+          </div>';
+      addMicrodata(divElement);
+      readyState = 'complete';
+      return new PageConfigResolver(gd).resolveConfig().then(config => {
+        expect(config.isLocked()).to.be.true;
+        expect(config.getProductId()).to.equal('pub1:premium');
```

Comment:
Hmm. I think the first one should win...

Reply:
The first one happens to be under "Section"

Reply:
Understood.